### PR TITLE
layer: Introduce basic layer/z-index support

### LINKED ISSUE #178: draw: Support layers
Support drawing behind already drawn nodes. This could be in the form of an `after(top, bottom)` function, that first draws `top`, then `bottom` but orders all drawing commands from bottom before the drawing commands of `top`.

Motivation: being able to draw behind elements while knowing their size/anchors.

### PR DESCRIPTION
**Pull request by johannes-wolf**
This allows setting a z-index on elements by wrapping them in `on-layer(index, body)`.

Fixes #178.

### CODE REVIEW

File: manual.typ [hunk 1]
```diff
@@ -441,6 +440,25 @@ for-each-anchor("demo", (name) => {
 })
 ```
 
+== Layers
+
+You can use layers to draw elements below or on top of other elements by using layers
+with a z-index. When rendering, all draw commands are sorted by their z-index (0 being
+the default).
```
Comment (fenjalien) [inline manual.typ:447]: Can we stick with the `layers` name? `z-index` could get confusing when we also allow a `z` component.

File: src/canvas.typ [hunk 2]
```diff
@@ -316,13 +316,17 @@
...
     }
-
   }
 
   if bounds == none {
     return []
   }
 
+  // Order draw commands by z-index
+  draw-cmds = draw-cmds.sorted(key: (cmd) => {
```
Comment (fenjalien) [inline src/canvas.typ:326]: This is a really simple way to do this, nice. But is it the best way given all drawables are looped even if the majority of them will be on the 0 layer?

File: src/draw.typ [hunk 3]
```diff
@@ -306,6 +306,30 @@
   ),)
 }
 
+/// Draw body on z-index layer
+///
+/// This can be used to draw elements behind already drawn elements by
+/// using a lower layer value (i.e -1).
+/// 
+/// - layer (integer, float): Z-Index to draw the children at
```
Comment (fenjalien) [inline src/draw.typ:314]: why a float?

File: src/draw.typ [hunk 4]
```diff
@@ -306,6 +306,30 @@
...
+/// - body (draw, function): Child elements or function (`ctx => draw-cmds`)
```
Comment (fenjalien) [inline src/draw.typ:315]: Do we have a dedicated type description to mean "elements", if not could you add one and use it in `canvas` and where ever else it needs.

File: src/draw.typ [hunk 5]
```diff
@@ -306,6 +306,30 @@
...
+/// - force (bool): Foce child elements on a different layer to this layer
```
Comment (fenjalien) [inline src/draw.typ:316]: I do not understand what this means

**Review by fenjalien — CHANGES_REQUESTED** [review]

File: src/draw.typ (hunk 3 above)
Comment (johannes-wolf) [reply src/draw.typ:314]: Why not?

File: src/canvas.typ (hunk 2 above)
Comment (johannes-wolf) [reply src/canvas.typ:326]: I see no problem with that.

File: src/draw.typ (hunk 5 above)
Comment (johannes-wolf) [reply src/draw.typ]: I removed that option. I was about moving from a layer to a different one, as the `layer` function does not overwrite an elements layer to support nesting.

File: src/draw.typ (hunk 4 above)
Comment (johannes-wolf) [reply src/draw.typ]: We use elements at other places, I changed it to that.

File: src/draw.typ (hunk 3 above)
Comment (fenjalien) [reply src/draw.typ]: To me layers should be discrete not continuous but I guess it doesn't really effect much.

File: manual.typ (hunk 1 above)
Comment (fenjalien) [reply manual.typ:447]: I don't see these changes?

File: src/canvas.typ (hunk 2 above)
Comment (fenjalien) [reply src/canvas.typ:326]: If you say so :)

File: manual.typ (hunk 1 above)
Comment (johannes-wolf) [reply manual.typ:447]: Oops, I had changed the code comment but missed the docs :)

**Pull request merged**